source/logger.c

```diff
@@ -213,29 +207,47 @@ static void s_threadsafe_log_call(napi_env env, napi_value node_log_fn, void *co
 
     /*
      * Look up `process` to use as this for the _rawDebug call, if these fail it's because the function
-     * call was queued during shutdown, so we will just skip out of here
+     * call was queued during shutdown, so we will just skip out of here.
+     * Avoid printing scary looking error messages (ex: avoid use of AWS_NAPI_CALL macro).
      */
     napi_value node_global = NULL;
-    AWS_NAPI_CALL(env, napi_get_global(env, &node_global), { goto done; });
+    if (napi_ok != napi_get_global(env, &node_global)) {
+        goto done;
+    }
+
     napi_value node_process = NULL;
-    AWS_NAPI_CALL(env, napi_get_named_property(env, node_global, "process", &node_process), { goto done; });
+    if (napi_ok != napi_get_named_property(env, node_global, "process", &node_process)) {
+        goto done;
+    }
+
+    for (struct aws_linked_list_node *list_node = aws_linked_list_rbegin(&msgs);
```

Comment:
I'd add a comment note for why reverse iteration is correct

Reply:
😱 OH NO THIS IS NOT CORRECT AT ALL
great catch. I have failed this interview